docs/topics/db/optimization.txt

```diff
@@ -411,4 +416,104 @@ objects to reduce the number of SQL queries. For example::
     my_band.members.add(me)
     my_band.members.add(my_friend)
 
-...where ``Bands`` and ``Artists`` have a many-to-many relationship.
+...where ``Band`` and ``Artist`` have a many-to-many relationship.
+
+If you have a list of objects, use the ``*``-operator when calling
+:meth:`~django.db.models.fields.related.RelatedManager.add`::
+
+    my_band.members.add(*new_members)
+
+Many source objects, many target objects
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When inserting pairs of objects into :class:`ManyToManyFields
+<django.db.models.ManyToManyField>`, use 
+:meth:`~django.db.models.query.QuerySet.bulk_create()` with multiple
+:attr:`through <django.db.models.ManyToManyField.through>` model instances 
+to reduce the number of SQL queries. For example::
+
+    Pizza_Topping = Pizza.toppings.through
```

Comment:
this model naming style may little confuse

Reply:
I'm open to other suggestions.